presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -125,6 +127,8 @@ public ExchangeClient(
             Executor pageBufferClientCallbackExecutor)
     {
         this.bufferCapacity = bufferCapacity.toBytes();
+        this.averageBytesPerRequest = this.bufferCapacity / INITIAL_REQUEST_COUNT;
```

Comment:
I wonder if it makes sense to limit the number of request.

The maximum response size possible is set by the `maxResponseSize`. Instead of introducing a fixed limit of request, how about we set the `averageBytesPerRequest` to the `maxResponseSize`?

Reply:
If it is still decided to limit the number of request - that feature will have to well tested and rolled out very carefully, as it may potentially increase query latency. 

With setting the `averageBytesPerRequest` to `maxResponseSize` in theory we should break nothing? However i would still recommend to have a safeguard for this change.

Reply:
I am not sure it makes sense to limit the number of requests. In the pathological case it may take a very long time for it to poll the nodes that have data.

In that same pathological case we shouldn't be polling nodes with the same frequency either.

I think what I am getting at is that limiting the number of nodes is correct except when it isn't and when we really do want to poll all of them unless we do something smarter. If the data is evenly distributed we should contact the "right" number of them anyways so why limit it manually?